src/network/url_downloader.cpp

```diff
@@ -134,7 +145,11 @@ auto get_header(QNetworkAccessManager* manager, const QUrl& url, const QNetworkR
     QTimer download_timeout;
     download_timeout.setInterval(timeout);
 
-    QNetworkRequest request{url};
+    QUrl adjusted_url{url};
+    if (adjusted_url.scheme() == "http")
+        adjusted_url.setScheme("https");
```

Comment:
Nice work on the `httpUrlBecomesHttps` test! 

It looks like that test only covers the "download" method, so this setScheme in the get_header method isn't being covered. Another test that tests get_header in test_url_downloader.cpp should be sufficient